src/main/java/frc/team2412/robot/commands/autonomous/AutonomousCommand.java

```diff
@@ -0,0 +1,93 @@
+        public static final TrapezoidProfile.Constraints kThetaControllerConstraints = new TrapezoidProfile.Constraints(
+                MaxAngularSpeedRadiansPerSecond, MaxAngularSpeedRadiansPerSecondSquared);
+
+        public static final double maxSpeedMetersPerSecond = 0.3;
+        public static final double maxAccelerationMetersPerSecondSquared = 0.1;
+        public static final double trackWidth = 1.0;
+        // Distance between centers of right and left wheels on robot
+        public static final double wheelBase = 1.0;
+        // Distance between front and back wheels on robot
+        public static final SwerveDriveKinematics driveKinematics = new SwerveDriveKinematics(
+                new Translation2d(wheelBase / 2, trackWidth / 2),
+                new Translation2d(wheelBase / 2, -trackWidth / 2),
+                new Translation2d(-wheelBase / 2, trackWidth / 2),
+                new Translation2d(-wheelBase / 2, -trackWidth / 2));
+    }
+    DrivebaseSubsystem drivebaseSubsystem;
+
+    public AutonomousCommand(DrivebaseSubsystem d) {
+        drivebaseSubsystem = d;
+    }
+
+    public Command getAutonomousCommand() {
+        // Create config for trajectory
+        TrajectoryConfig config = new TrajectoryConfig(
+                AutoConstants.maxSpeedMetersPerSecond,
+                AutoConstants.maxAccelerationMetersPerSecondSquared)
+                        // Add kinematics to ensure max speed is actually obeyed
+                        .setKinematics(AutoConstants.driveKinematics);
+        // creating trajectory path (right now is a square)
+        Trajectory exampleTrajectory = TrajectoryGenerator.generateTrajectory(
+                new Pose2d(0, 0, Rotation2d.fromDegrees(0)),
+                List.of(new Translation2d(1, 0), new Translation2d(1, 1), new Translation2d(0, 1)),
+                new Pose2d(0, 0.0, Rotation2d.fromDegrees(0)),
+                config);
+        // creates thetacontroller (rotation)
+        ProfiledPIDController thetaController = new ProfiledPIDController(
+                0.000000005, 0, 0, AutoConstants.kThetaControllerConstraints);
+        thetaController.enableContinuousInput(-Math.PI, Math.PI);
+
+        SwerveControllerCommand swerveControllerCommand = new SwerveControllerCommand(
+                exampleTrajectory,
+                drivebaseSubsystem::getPoseAsPoseMeters, // Functional interface to feed supplier
+                AutoConstants.driveKinematics,
+
+                // Position controllers
+                new PIDController(AutoConstants.PXController, 0, 0),
+                new PIDController(AutoConstants.PYController, 0, 0),
+                thetaController,
+                drivebaseSubsystem::updateModules,
+                drivebaseSubsystem);
+
+
+        // Reset odometry to the starting pose of the trajectory.
+        drivebaseSubsystem.resetPose(exampleTrajectory.getInitialPose());
```

Comment:
To instead have the path start form the current position, do something like `exampleTrajectory.relativeTo(drivebaseSubsystem.getPoseAsPoseMeters())` above the creation of `swerveControllerCommand` and then delete the `resetPose` call here

Reply:
did it